jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/StatisticsHandler.java

```diff
@@ -175,9 +111,93 @@ private long spentTimeNs()
         @Override
         public void process(Request ignored, Response response, Callback callback) throws Exception
         {
-            _processStats.increment();
             _processStartTimeStamp = System.nanoTime();
-            super.process(this, response, callback);
+            _processStats.increment();
+            _requestStats.increment();
+
+            _connectionStats.computeIfAbsent(getConnectionMetaData().getId(), id ->
+            {
+                // TODO test this with localconnector endpoint that has multiple requests per connection.
+                getConnectionMetaData().getConnection().addEventListener(new Connection.Listener()
+                {
+                    @Override
+                    public void onClosed(Connection connection)
+                    {
+                        // complete connections stats
+                        _connectionStats.remove(id);
+                    }
+                });
+                return "SomeConnectionStatsObject";
```

Comment:
Can this be a `Set`? If not, use better constant like `static final Object PRESENT = new Object()`.